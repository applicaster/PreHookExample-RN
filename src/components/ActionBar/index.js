import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import EventDetailCount from '../EventDetailCount';
import ShareButton from '../ShareButton';
import TwitterActionButtons from '../TwitterActionButtons';
import { styles } from './style';

class ActionBar extends Component {
  renderShareButton() {
    return (<View style={styles.shareButtonContainer}>
      <ShareButton message={this.props.textToShare} />
    </View>);
  }
  
  renderInstagramBar() {
    const { likesCount, commentsCount, eventId, eventOriginUrl } = this.props;
    return (
      <View style={[styles.buttonBar]}>
        <EventDetailCount key={'likes'} eventOriginUrl={eventOriginUrl} openOriginUrl eventId={eventId} count={likesCount} label={'Likes'} />
        <EventDetailCount key={'comments'} eventOriginUrl={eventOriginUrl} openOriginUrl eventId={eventId} count={commentsCount} label={'Comments'} />
        {this.renderShareButton()}
      </View>
    );
  }

  renderTwitterBar() {
    const { likesCount, commentsCount, eventId, retweetCount } = this.props;
    return (
      <View style={[styles.buttonBar]}>
        <TwitterActionButtons
          eventId={eventId}
          favoriteCount={likesCount}
          replyCount={commentsCount}
          retweetCount={retweetCount}
        />
        {this.renderShareButton()}
      </View>
    );
  }

  renderFacebookBar() {
    const { likesCount, commentsCount, eventId } = this.props;
    return (
      <View style={[styles.buttonBar]}>
        <EventDetailCount key={'likes'} eventId={eventId} count={likesCount} label={'Likes'} />
        <EventDetailCount key={'comments'} eventId={eventId} count={commentsCount} label={'Comments'} />
        {this.renderShareButton()}
      </View>
    );
  }

  render() {
    switch (this.props.socialNetwork) {
      case 'instagram':
        return this.renderInstagramBar();
      case 'facebook':
        return this.renderFacebookBar();
      case 'twitter':
        return this.renderTwitterBar();
      default:
        return null;
    }
  }
}

ActionBar.propTypes = {
  commentsCount: PropTypes.number,
  eventId: PropTypes.string,
  eventOriginUrl: PropTypes.string,
  socialNetwork: PropTypes.string,
  likesCount: PropTypes.number,
  commentsCount: PropTypes.number,
  textToShare: PropTypes.string,
};

ActionBar.contextTypes = {
  navigation: PropTypes.object,
};

export default ActionBar;
