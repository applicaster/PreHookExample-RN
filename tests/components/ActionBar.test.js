import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ActionBar from '../../src/components/ActionBar';

jest.mock('@applicaster/feed-rn-utils', () => 'FeedRNUtils');
jest.mock('../../src/components/TwitterActionButtons', () => 'TwitterActionButtons');
jest.mock('../../src/components/FacebookActionButtons', () => 'FacebookActionButtons');
jest.mock('../../src/components/InstagramActionButtons', () => 'InstagramActionButtons');
jest.mock('../../src/components/ShareButton', () => 'ShareButton');

describe('ActionBar', () => {
  test('component renders correctly for instagram', () => {
    const html = renderer.create(
      <ActionBar socialNetwork={'instagram'} likesCount={1} commentsCount={3} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
  
  test('component renders correctly for facebook', () => {
    const html = renderer.create(
      <ActionBar socialNetwork={'instagram'} likesCount={1} commentsCount={3} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly for twitter', () => {
    const html = renderer.create(
      <ActionBar socialNetwork={'instagram'} likesCount={1} commentsCount={3} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly for no social network', () => {
    const html = renderer.create(
      <ActionBar socialNetwork={null} likesCount={null} commentsCount={null} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
