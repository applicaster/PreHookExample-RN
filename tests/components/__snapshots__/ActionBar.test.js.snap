// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar component renders correctly for instagram 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "height": 30,
        "paddingBottom": 10,
        "paddingHorizontal": 7,
      },
      Object {
        "marginLeft": 3,
      },
    ]
  }
>
  <EventDetailCount
    count={1}
    eventId={undefined}
    label="Likes"
  />
  <EventDetailCount
    count={3}
    eventId={undefined}
    label="Comments"
  />
  <View
    style={
      Object {
        "bottom": 10,
        "position": "absolute",
        "right": 7,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "uri": "feed_rn_share_button",
          }
        }
        style={
          Array [
            Object {
              "height": 30,
              "width": 30,
            },
            Object {
              "tintColor": "rgba(255,255,255, 0.60)",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`ActionBar component renders correctly for no social network 1`] = `null`;
